### ISSUE TITLE
Media Panel: custom x-axis options

### ISSUE THREAD

**Issue by tshrjn** [wandb/wandb#1093]
### Description
The step counter for [custom plots](https://docs.wandb.com/library/log#custom-plots) doesn’t have other `x-axis` option except for `step`.
And the Legend doesn’t work properly with custom plots. [The color doesn’t show up here.]
Also, the title set up in matplotlib cuts off.

<img width="1565" alt="Screen Shot 2020-06-10 at 3 40 36 AM" src="https://user-images.githubusercontent.com/8372098/84241395-68b60b80-aacd-11ea-8c23-f40262566d18.png">

### Code Description
Reference code being used in pytorch-lighning's `LightningModule`.
```
        fig, ax = plt.subplots()
        colors = {
            'train': 'blue',
            'valid': 'green',
            'test':  'red'}

        for split, yearwise_skills in skill_dict.items():
            years, skills = list(zip(*yearwise_skills))
            ax.scatter(years, skills, c=colors[split], label=split)

        ax.set_title(f'Epoch {self.current_epoch}')
        ax.set_ylabel(f'Skills at Epoch {self.current_epoch}')
        ax.set_xlabel('Years')
        ax.legend()
        ax.grid(True)

        self.logger.experiment.log({"Skill_by_years": plt})
```

### Version info

```bash
* CUDA:
        - GPU:
                - GeForce RTX 2080 Ti
        - available:         True
        - version:           10.1
* Packages:
        - numpy:             1.18.1
        - pyTorch_debug:     False
        - pyTorch_version:   1.5.0
        - pytorch-lightning: 0.7.5
        - tensorboard:       2.2.0
        - tqdm:              4.45.0
        - wandb:            0.8.35
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                -
        - processor:         x86_64
        - python:            3.6.10
        - version:           #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019
```

**Comment by tshrjn** [comment]
Is there a way to use the step slider with Plotly charts? 

Also, a use case for step-slider is similar to that of the x-axis available in the workspace for instance, epoch instead of a step.

**Comment by vanpelt** [comment]
Yep, if you call wandb.log multiple times the step slider will be available for plotly charts.  We'll add a +1 to custom x-axis for media panels.

**Comment by tshrjn** [comment]
Btw, How can we send plotly charts to `wandb.log()`? Is there an example? I couldn't find any on the [docs](https://docs.wandb.com/library/log#custom-plots)

**Comment by vanpelt** [comment]
Sorry about that, we'll update the docs.  You can pass in a Plotly figure object to wandb.log.  Any of the examples here: https://plotly.com/python/line-charts/ would work by replacing `fig.show()` with `wandb.log({"plotly_chart": fig})`

**Comment by ariG23498** [comment]
Hey folks
In the past few months we've majorly reworked the CLI and UI for Weights & Biases. We're closing stale issues. Please comment to reopen. :smile:

**Comment by ai-alanov** [comment]
Please reopen the issue. The custom x-axis option is really needed. For example, I save images per epoch and the ability to change the x-axis from step to epoch would be great!

**Comment by elisim** [comment]
Hi, how to change the x-axis from step to epoch?

**Comment by lely475** [comment]
Would appreciate the custom x-axis option as well, I save multiple types of images per epoch and it is a bit messy using step.

**Comment by bzrry** [comment]
Agreed, it would be much easier to gauge model progress if we could see images labeled by epoch rather than step.

**Comment by azinonos** [comment]
Agree as well, I see this has been raised over a year ago but hasn't been implemented yet?

**Comment by LucasZhan** [comment]
Agree with the need of custom x-axis for media step-slider.

**Comment by armanhar** [comment]
Thanks for the feedback @LucasZhan, I've bumped up the priority for the ticket.

**Comment by KacperKazan** [comment]
Hi, I've also arrived here by searching how to log media objects by epoch rather than step. This custom x-axis feature in the media panel would really be useful!

**Comment by armanhar** [comment]
Thanks for writing in @KacperKazan, I'll notify you once we have any updates on this.

**Comment by dongzhuoyao** [comment]
really need it, pls prioritize this feature.

**Comment by tlpss** [comment]
I would love to have this feature as well @armanhar

**Comment by vincentX3** [comment]
Still waiting for this feature. Pls.

**Comment by AndreasKernbach** [comment]
I would love to have that feature, still waiting for it.

**Comment by vincentX3** [comment]
已收到，稍后就看。

**Comment by vincentX3** [comment]
已收到，稍后就看。

**Comment by vincentX3** [comment]
已收到，稍后就看。

**Comment by adrialopezescoriza** [comment]
Please add the feature, would be really useful

**Comment by vincentX3** [comment]
已收到，稍后就看。

**Comment by tristan-spaulding2** [comment]
Hi all--we've just started to roll out support for custom steps in our media panels, first to our personal and academic users.    This is live on our SaaS environment now--take a look and let us know if this helps.  Thanks.

![Image](https://github.com/user-attachments/assets/b1838cc9-9ec0-491c-8bfd-49ed354c4f1f)

**Comment by vincentX3** [comment]
已收到，稍后就看。
